File: src/main/java/ftc/shift/sample/models/Subscription.java
package ftc.shift.sample.models;

import javax.persistence.*;

public class Subscription {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    long id;

    @ManyToOne(optional=false, cascade= CascadeType.ALL)
    @JoinColumn (name="my_user_id")
    User user;

    @ManyToOne(optional=false, cascade= CascadeType.ALL)
    @JoinColumn (name="my_group_id")
    Group group;
}

File: src/main/java/ftc/shift/sample/models/Subscription.kt
package ftc.shift.sample.models

import javax.persistence.*


@Entity
data class Subscription(@ManyToOne @JoinColumn(name = "user_id", referencedColumnName = "id")
                        val user: User? = null,
                        @ManyToOne @JoinColumn(name = "group_id", referencedColumnName = "id")
                        val group: Group? = null,
                        @OneToOne @JoinColumn(name = "target_id", referencedColumnName = "id")
                        var target: User? = null,
                        @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
                        val id: Long = 0)